src/mappings/rewardsDistributor.ts

```diff
@@ -129,6 +159,90 @@ export async function handleRewardsClaimed(
   } catch {
     logger.error(`ERROR: handleRewardsClaimed`);
   }
+
+  await updateEraRewardClaimed(event);
+}
+
+interface EraRewardData {
+  indexerId: string;
+  delegatorId: string;
+  eraId: string;
+  isCommission: boolean;
+  claimed: boolean;
+  amount?: bigint;
+  createdBlock?: number;
+  createdTimestamp?: Date;
+}
+
+async function updateEraReward(data: EraRewardData): Promise<EraReward> {
+  console.log('updateEraReward', data);
+
+  const id = `${data.indexerId}_${data.delegatorId}_${data.eraId}${
+    data.isCommission ? '_commission' : ''
+  }`;
+
+  let eraReward = await EraReward.get(id);
```

Comment:
You should know if you want to create or update `EraReward`, you can gain some performance by skipping this step.

Reply:
should I consider if it's doing regenerating database from the chain and validate before creating the records?

Reply:
reindex will always start from empty database